src/pages/member/UserEvents.js

```diff
@@ -0,0 +1,292 @@
+    // filter events by the date
+    const now = new Date()
+    return AllEventCards.filter((event) => event.startDate && new Date(event.startDate) < now)
+  }, [AllEventCards, selectedPanel])
+
+  return (
+    <div>
+      <Helmet>
+        <title>Biztech User Events Dashboard</title>
+      </Helmet>
+      <div style={styles.flex}>
+        <div style={styles.sidePanel.layout}>
+          <Typography variant='h1' style={styles.header}>Events</Typography>
+          <List>
+            <ListItem
+              style={selectedPanel === PANEL_STATES.FAVORITES
+                ? styles.sidePanel.activeButton
+                : styles.sidePanel.button}
+              onClick={() => handlePanelChange(PANEL_STATES.FAVORITES)}
+              button
+            >
+              <ListItemText><StarBorder fontSize='small' />&nbsp;Favorites</ListItemText>
+            </ListItem>
+            <ListItem
+              style={selectedPanel === PANEL_STATES.REGISTERED
+                ? styles.sidePanel.activeButton
+                : styles.sidePanel.button}
+              onClick={() => handlePanelChange(PANEL_STATES.REGISTERED)}
+              button
+            >
+              <ListItemText><PlaylistAddCheck fontSize='small' />&nbsp;Registered</ListItemText>
+            </ListItem>
+            <ListItem
+              style={selectedPanel === PANEL_STATES.ALL
+                ? styles.sidePanel.activeButton
+                : styles.sidePanel.button}
+              onClick={() => handlePanelChange(PANEL_STATES.ALL)}
+              button
+            >
+              <ListItemText><Search fontSize='small' />&nbsp;All</ListItemText>
+            </ListItem>
+          </List>
+        </div>
+        <div style={styles.tabs.container}>
+
+          <Tabs
+            value={tabIndex}
+            indicatorColor='primary'
+            textColor='primary' // TODO: Set the primary color theme
+            onChange={handleTabChange}
+            style={styles.tabs.layout}
+          >
+            <Tab label='Upcoming' style={styles.tabs.tab} />
+            <Tab label='Past' style={styles.tabs.tab} />
+            <Tab label='All' style={styles.tabs.tab} />
+          </Tabs>
+
+          <EventPanel currentIndex={tabIndex} index={TAB_STATES.UPCOMING}>
+            <div style={styles.rows}>
+              {UpcomingEventCards.map((event) => (
+                <EventCard
+                  event={event}
+                  key={event.id}
+                  variant='user'
+                  favorited={eventsFavoritedIds.includes(event.id)}
+                  handleCardClick={redirectToEvent}
+                  handleFavorite={handleFavoriteEvent}
+                  cardStyle={{ width: '40%' }}
+                />
+              ))}
+            </div>
+          </EventPanel>
+          <EventPanel currentIndex={tabIndex} index={TAB_STATES.PAST}>
+            <div style={styles.rows}>
+              {PastEventCards.map((event) => (
```

Comment:
could we check the index here to decide which array to use? so we don't have to have duplicate as much code?

Reply:
Hm.. Does turning it into its own function help?

Reply:
yeah with like 3 if statements?